lib/class-wp-rest-menus-controller.php

```diff
@@ -13,6 +13,46 @@
  */
 class WP_REST_Menus_Controller extends WP_REST_Terms_Controller {
 
+	/**
+	 * Checks if a request has access to read terms in the specified taxonomy.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool|WP_Error True if the request has read access, otherwise false or WP_Error object.
+	 */
+	public function get_items_permissions_check( $request ) {
+		$tax_obj = get_taxonomy( $this->taxonomy );
+		if ( ! $tax_obj || ! $this->check_is_taxonomy_allowed( $this->taxonomy ) ) {
+			return false;
+		}
+		if ( ! current_user_can( $tax_obj->cap->edit_terms ) ) {
+			if ( 'edit' === $request['context'] ) {
+				return new WP_Error( 'rest_forbidden_context', __( 'Sorry, you are not allowed to edit terms in this taxonomy.' ), array( 'status' => rest_authorization_required_code() ) );
+			}
+			return new WP_Error( 'rest_forbidden_access', __( 'Sorry, you view these menus, unless you have access to allow edit them. ' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+		return true;
+	}
+
+	/**
+	 * Checks if a request has access to read or edit the specified menu.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool|WP_Error True if the request has read access for the item, otherwise false or WP_Error object.
+	 */
+	public function get_item_permissions_check( $request ) {
+		$term = $this->get_term( $request['id'] );
+		if ( is_wp_error( $term ) ) {
+			return $term;
+		}
+		if ( ! current_user_can( 'edit_term', $term->term_id ) ) {
```

Comment:
This is a copy and paste from [here](https://github.com/WordPress/WordPress/blob/aa1fdcbd52281316ab82435ec42000ec64deea95/wp-includes/rest-api/endpoints/class-wp-rest-terms-controller.php#L341-L351)

Reply:
So should `get_items_permissions_check` be adjusted? Or is that also sourced from somewhere?

Reply:
I think there is a good reason why it was done this way in the term controller. I think we should leave it as is, as it works. `edit_term` is mapped correctly in [map_meta_cap](https://developer.wordpress.org/reference/functions/map_meta_cap/)

Reply:
Yeah, I was confused why the mismatch between `get_item` and `get_items`, but that might also be a brainfart.

Reply:
@TimothyBJacobs `get_items()` can't be changed to just use `edit_terms` because `map_meta_cap()` has no context about the taxonomy to check for. Unlike with `edit_term`, where it can fetch the taxonomy object based on the provided term ID.

Reply:
@ocean90 This PR is blocked, can you create a new issue.